### DISCUSSION TITLE
Too Many Requests to account.xiaomi.com

### DISCUSSION THREAD

**Discussion by hasquid** [Tasshack/dreame-vacuum#220] [General]
I have been monitoring network traffic and noticed that enabling the integration leads to repeated requests to account.xiaomi.com. 

The requests seem to be every 10 seconds, and do not seem to stop at any time:

![image](https://github.com/Tasshack/dreame-vacuum/assets/63639063/bf18e7ca-7c80-4f9e-9650-b66a665c9783)

The first request receives an IP, and the second one receives no data.

Blocking the entire domain does not limit the functionality of the integration from what I observed so far. But can you confirm if this behavior is expected?

**Comment by Tasshack** [comment]
In order to get the map updates, integration needs to constantly check new object name from the cloud. If you block the request and make any changes on map from the app (including changing the current floor), you will see that change won't be reflected in Home Assistant. Unfortunately this is a limitation of the Xiaomi Cloud, there is no other way to get the map changes and this is exactly how it is implemented on the Mi Home app too.
They have fixed this on the Dreamehome cloud and added a mqtt server in order to get changes from the cloud without needing to send requests to the cloud constantly. If you have a vacuum that can be installed on Dreamehome app, you will see that the integration does not send requests to the cloud when it is installed on Dreamehome cloud.
I knew this could be an issue because of that i have made period of the requests are dynamic and minimum time is 10 seconds when the device is idle. If you won't request the map image long enough you should observe that the request interval will be longer.
